src/VecSim/vec_sim.cpp

```diff
@@ -190,6 +190,23 @@ extern "C" VecSimQueryResult_List VecSimIndex_TopKQuery(VecSimIndex *index, cons
     return results;
 }
 
+extern "C" VecSimQueryResult_List VecSimIndex_RangeQuery(VecSimIndex *index, const void *queryBlob,
+                                                         float radius,
+                                                         VecSimQueryParams *queryParams,
+                                                         VecSimQueryResult_Order order) {
+    assert((order == BY_ID || order == BY_SCORE) &&
+           "Possible order values are only 'BY_ID' or 'BY_SCORE'");
+    assert(radius >= 0 && "radius must be non-negative");
+    VecSimQueryResult_List results = index->rangeQuery(queryBlob, radius, queryParams);
+
+    if (order == BY_SCORE) {
+        sort_results_by_score(results);
+    } else {
+        sort_results_by_id(results);
+    }
```

Comment:
This could wait for the HNSW PR, but I think that HNSW algorithm will return results ordered by score anyway, so maybe do the sorting in the specific algorithm function.

Reply:
Actually I'm not sure, because in the range query implementation for HNSW, we would probably *not* use a heap sorted by the scores as the `top_candidates` set, since we will not need to remove vectors from it (if you're in the range - you're in), unlike in KNN.